### PR TITLE
feat(elbv2): support pathpattern array

### LINKED ISSUE #6497: The pathPattern property, in AddApplicationTargetsProps and AddApplicationTargetGroupsProps, should accept a list, not just a simple string.
There is currently no way to add a target group to `ApplicationListener` using multiple path patterns.

According to the doc, https://docs.aws.amazon.com/elasticloadbalancing/latest/application/load-balancer-listeners.html#path-conditions, one should be able to use one or more path patterns as a set. I think this can be very useful in general, but it is also required to be able to support certain simple (but important) use cases. CDK currently does not support this in `ApplicationListener.addTargets()` and `ApplicationListener.addTargetGroups()`.

As an example, currently it is not possible to add a target group with path patterns, "/extra" and "/extra/\*". (There does not seem to be a way to express these two as a single path pattern.) If you specify "/extra", then it'll match "/extra" but not "/extra/abc", ... If you specify "/extra/\*", then it'll match patterns like "/extra/xyz", ... but not "/extra".

I would consider it a fundamental limitation of the API ("bug") rather than a missing feature. (I haven't found a workaround for this use case.)

### Reproduction Steps

1. Create an `ApplicationLoadBalancer`.
1. Add an `ApplicationListener` to the ALB.
1. Create a new target group.
1. Try to add the new TG to path ["/extra", "/extra/\*"] using `ApplicationListener.addTargetGroups()`.

### Environment

  - **CLI Version     : 1.25.0 (build 5ced526)
  - **Language         : Typescript

--- 

This is :bug: Bug Report

### PR DESCRIPTION
**Pull request by DerkSchooltink**
Adding support for an array of path patterns for application listener rule config.

Closes #6497

Usage:
```
        listener.addTargets("id", {
            priority: 1,
            pathPattern: ["/path/1/", "/path/2/"],
            port: 8080,
            deregistrationDelay: Duration.minutes(1),
            targets: [target]
        });
```
or a single pattern (so no breaking changes):

```
        listener.addTargets("id", {
            priority: 1,
            pathPattern: "/path/1/",
            port: 8080,
            deregistrationDelay: Duration.minutes(1),
            targets: [target]
        });
```

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by DerkSchooltink** [comment]
Sorry for the build spam I had to fix some minor things regarding README's and tests

**Comment by realharry** [comment]
"Unofficial" LGTM. ;)

**Comment by realharry** [comment]
The only thing I don't know is, do all APIs have to be consistent across different language implementations? I've only been using typescript CDK, but I presume they are not necessarily all the same in terms of API design?

File: packages/@aws-cdk/aws-elasticloadbalancingv2/lib/alb/application-listener-rule.ts [hunk 1]
```diff
@@ -60,7 +60,7 @@ export interface BaseApplicationListenerRuleProps {
    *
    * @default - No path condition.
    */
-  readonly pathPattern?: string;
+  readonly pathPattern?: string | string[];
```
Comment (rix0rrr) [inline packages/@aws-cdk/aws-elasticloadbalancingv2/lib/alb/application-listener-rule.ts]: No union types please. The fact that they are currently supported by jsii is a mistake and we will take them out eventually.

Add a new field named `pathPatterns?: string[]` and add validation to make sure they're not both set.

Comment (rix0rrr) [reply packages/@aws-cdk/aws-elasticloadbalancingv2/lib/alb/application-listener-rule.ts]: I'm still seeing the old commits. Did you push them?

**Comment by rix0rrr** [comment]
> The only thing I don't know is, do all APIs have to be consistent across different language implementations? I've only been using typescript CDK, but I presume they are not necessarily all the same in terms of API design?

They are, or as near as possible. The cross-language bindings are automatically generated.

**Comment by DerkSchooltink** [comment]
Forgot to update the README, this is fixed now.

**Pull request merged**